### DISCUSSION TITLE
URL::to and url() question regarding the query string

### DISCUSSION THREAD

**Discussion by EmranMR** [laravel/framework#45413] [Q&A] [answered]
Hi,

Laravel v9.45.1:-

I am not sure if I am confused or this is a bug.
I am trying to generate a link to an external website with query strings.url helper function has an argument called $parameter and URL::to has an argument called $extra. I presumed these are query string bags same as the ones in the route() helper function.

Now when I add my query string the generated url just ignores them all... If I use a relative  path, the parameters are added as an additional path rather than query string. Attached some photos from the tinker.

Is there a way to do this for absolute URLs? 

Thanks!

<img width="537" alt="Screenshot 2022-12-23 at 14 25 14" src="https://user-images.githubusercontent.com/11975985/209352584-e10f290f-e73e-4b50-9027-e8fc31efd849.png">

<img width="620" alt="Screenshot 2022-12-23 at 14 25 43" src="https://user-images.githubusercontent.com/11975985/209352606-c332dd92-529a-49e5-8127-87f724e4c79b.png">

<img width="620" alt="Screenshot 2022-12-23 at 14 33 14" src="https://user-images.githubusercontent.com/11975985/209353154-1290be6f-7376-4231-935b-35996a5474bd.png">
<img width="620" alt="Screenshot 2022-12-23 at 14 32 43" src="https://user-images.githubusercontent.com/11975985/209353156-48fc55ce-12e4-4b2d-8616-729a27027443.png">

**Comment by hafezdivandari** [comment, marked as answer]
The `url` helper function and `URL` facade are for generating URLs within your application (internal) you may use PHP `http_build_query` function instead:

```php
$url = 'https://www.google.com?'.http_build_query(['bar' => 'baz']); // https://www.google.com?bar=baz
```

additionally, if you want to [redirect to an external domain](https://laravel.com/docs/9.x/responses#redirecting-external-domains) you may use:

```php
return redirect()->away('https://www.google.com?'.http_build_query(['bar' => 'baz']));
```

**Reply by EmranMR** [reply]
Ah thanks! I got fed up and made my own helper using the first method.
However thanks for the second snippet as I wasn't aware of the `redirect()->away`!  I would be playing around with that, from now on for sure! 🙏
